libraries/cms/application/site.php

```diff
@@ -651,6 +621,25 @@ protected function initialiseApp($options = array())
 	}
 
 	/**
+	 * Language check
+	 *
+	 * @param  string  Language to check.
+	 *
+	 * @return  mixed  string  The passed language if it is an available language.
+	 *                 null    If the language is not an available language.
+	 *
+	 * @since   3.4.2
+	 */
+	protected function checkLang($lang)
```

Comment:
take that argument up with @bakual for example, but basically all methods and functions, regardless of them being protected or not, have been declared B/C at one time or another. If there was no problem removing/changing protected methods, I could simply rewrite the Joomla router from scratch instead of jumping through hundreds of hoops here.